lib/ret_web/controllers/api/v1/account_controller.ex

```diff
@@ -30,6 +30,17 @@ defmodule RetWeb.Api.V1.AccountController do
     exec_api_create(conn, params, @record_schema, &process_account_update_record/2)
   end
 
+  def delete(conn, params) do
+    current_account = Guardian.Plug.current_resource(conn)
+    account_to_delete = Ret.Account.get_account_by_id(params[:account_id])
+
+    case Ret.AccountContext.delete_account(current_account, account_to_delete) do
```

Comment:
What if `delete_account/2` took the id of the account to delete, rather than the data structure? Then there would only be one domain call instead of two.

Reply:
This would obviate `Account.get_account_by_id/1` entirely.

Reply:
I think I'll stick to passing structs around, since that's the pattern we generally use across the codebase. Guardian's current_resource is already an Account struct. The Canada authorization rule also requires a struct.